### PR TITLE
fix(redirects): attempt to get only params in dev mode

### LINKED ISSUE #8639: Dynamic routes in Astro `redirects` error
### Astro Info

```block
(run on StackBlitz)

Astro                    v3.1.2
Node                     v16.20.0
System                   Linux (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             none
```

### If this issue only occurs in one browser, which browser is a problem?

n/a

### Describe the Bug

Astro supports a [`redirects`](https://docs.astro.build/en/core-concepts/routing/#redirects) field in `astro.config`. According to our docs this should be able to support dynamic routes:

> Dynamic routes are allowed as long as both the new and old routes contain the same parameters

I’m seeing errors for each of the following cases:
```js
  redirects: {
    '/old/[...spread]': '/new/[...spread]',
    '/old/[dynamic]': '/[dynamic]',
    '/old/[dynamic]/[route]': '/[dynamic]/[route]',
  }
```

In each case, attempting to load a route from one of the `/old/` paths, throws:

> `getStaticPaths()` function is required for dynamic routes. Make sure that you `export` a `getStaticPaths` function from your dynamic route.

The dynamic routes at the new location each contain `getStaticPaths()`.

### What's the expected result?

Dynamic redirects containing the same parameters redirect to the new location as expected.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-gjbu5b?file=astro.config.mjs

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by lilnasy**
## Changes
- Fixes #8639 

## Testing
Adapted from repro.

## Docs
Does not affect usage.

### CODE REVIEW

**Review by ematipico — APPROVED** [review]
Plain and simple 👊👌

File: packages/astro/test/redirects.test.js [hunk 1]
```diff
@@ -170,11 +182,27 @@ describe('Astro.redirect', () => {
...
+
+			it('performs dynamic redirects with multiple params', async () => {
+				const response = await fixture.fetch('/more/old/hello/world', { redirect: 'manual' });
+				expect(response.headers.get('Location')).to.equal('/more/hello/world');
+			});
+			
+			it.skip('falls back to spread rule when dynamic rules should not match', async () => {
+				const response = await fixture.fetch('/more/old/welcome/world', { redirect: 'manual' });
+				expect(response.headers.get('Location')).to.equal('/more/new/welcome/world');
 			});
```
Comment (lilnasy) [inline packages/astro/test/redirects.test.js:206]: I'll let someone else tackle this.

**Review by matthewp — APPROVED** [review]
Looks great

**Comment by lilnasy** [comment]
We don't have a system setup for that case. This is a small change, however. I'd recommend publishing an npm package for yourself with `dist/core/render/params-and-props.js` edited.
```
"dependencies": {
  "astro": "npm:felixfrey-astro@2.10.16"
}
```

**Pull request merged**